### PR TITLE
add scalex

### PR DESCRIPTION
**Pull request by jsxlei**
### Mandatory

Name of the tool: SCALEX

Short description: SCALEX is an integration and projection tool for atlas-level single-cell RNA-seq and ATAC-seq data.

How does the package use scverse data structures (please describe in a few sentences): 
This package anndata and muon packages as data structure and use scanpy for preprocessing and visualization.

-   [x] The code is publicly available under an [OSI-approved](https://opensource.org/licenses/alphabetical) license
-   [x] The package provides versioned releases
-   [x] The package can be installed from a standard registry (e.g. PyPI, conda-forge, bioconda)
-   [x] The package uses automated software tests and runs them via continuous integration (CI)
-   [x] The package provides API documentation via a website or README
-   [x] The package uses scverse datastructures where appropriate (i.e. AnnData, MuData or SpatialData and their modality-specific extensions)
-   [x] I am an author or maintainer of the tool and agree on listing the package on the scverse website

### Recommended

-   [ ] Please announce this package on scverse communication channels (zulip, discourse, twitter)
-   [x] Please tag the author(s) these announcements. Handles (e.g. `@scverse_team`) to include are:
    -   Twitter: @jsxlei
    -   Zulip:
    -   Discourse:
    -   Mastodon:
-   [x] The package provides tutorials (or "vignettes") that help getting users started quickly
-   [ ] The package uses the [scverse cookiecutter template](https://github.com/scverse/cookiecutter-scverse).

### CODE REVIEW

**Comment by canergen** [comment]
Hi, thanks for contributing. It was helpful to scVI-tools in two regards (more benchmarking of not encoding covariates and conditional batchnorm).
I'm somewhat confused by the statement in the original paper:

_However, the conditional VAE design of scVI requires model augmentation and retraining when integrating new data, meaning that scVI is not an online method._

This is actually not true and encode_covariates is False by default.

To reduce package redundancy and rely on scverse core packages (as well as support future improved scverse dataloaders): Would you be open to contribute SCALEX as either an scvi-tools external package or as a tutorial to scvi-tools. We have an open PR for conditional batchnorm and my first understanding is that when this is merged SCALEX can be used for each scvi-tools model by setting the hyperparameters accordingly.

**Comment by martinkim0** [comment]
Hi @jsxlei, thanks for your contribution! Just a quick edit: looks like the repo doesn't depend on the scverse cookiecutter template, so I'm going to unmark that checkbox.

**Comment by gtca** [comment]
Thanks for contributing, @jsxlei!

As a quick note — I think there are actually no tests or CI that I can find in the repository at the moment. I'll unmark that checkbox for now as well.

**Comment by jsxlei** [comment]
> Hi @jsxlei, thanks for your contribution! Just a quick edit: looks like the repo doesn't depend on the scverse cookiecutter template, so I'm going to unmark that checkbox.

Thank you for your response. I will 

> Hi @jsxlei, thanks for your contribution! Just a quick edit: looks like the repo doesn't depend on the scverse cookiecutter template, so I'm going to unmark that checkbox.

Thank you for your response. I will try to use the exact template. What else to do to be added to the scverse? Thanks!

**Comment by jsxlei** [comment]
> Thanks for contributing, @jsxlei!
> 
> As a quick note — I think there are actually no tests or CI that I can find in the repository at the moment. I'll unmark that checkbox for now as well.

Thank you for pointing out! I will add CI soon.

**Comment by jsxlei** [comment]
> Hi, thanks for contributing. The publication was helpful to me for scVI in two regards (more benchmarking of not encoding covariates and conditional batchnorm). I'm somewhat confused by the statement in the original paper:
> 
> _However, the conditional VAE design of scVI requires model augmentation and retraining when integrating new data, meaning that scVI is not an online method._
> 
> This is actually not true and encode_covariates is False by default.
> 
> (Edited here as the original language was confusing) As a suggestion: Would you be open to contribute SCALEX as either an scvi-tools external package or as a tutorial to scvi-tools. We have an open PR for conditional batchnorm and my first understanding is that when this is merged SCALEX can be used for each scvi-tools model by setting the hyperparameters accordingly.

Hi, Thank you for your suggestions. I am quite open to scVI-tools. How can we make forward on this. (As for the claimed in the paper, it was compared with original scVI in the first paper. We also compared with scVI without the condition in the encoder in our paper.)

**Comment by jsxlei** [comment]
> Hi @jsxlei, thanks for your contribution! Just a quick edit: looks like the repo doesn't depend on the scverse cookiecutter template, so I'm going to unmark that checkbox.

Hi @martinkim0 , I just add the pytest part and follow the cookiecutter template as much as I can including the CI. Danila has helped me checked it. Could you please review and approve the merge when at your convenience. Thanks!

**Review by gtca — APPROVED** [review]
LGTM

**Comment by gtca** [comment]
Hey @jsxlei, thanks for the updates!

Just to provide more suggestions for improvements:

- [ ] fix the Markdown syntax in your [README.md](https://github.com/jsxlei/SCALEX/blob/main/README.md) (it is not rendered properly [on the project's page on PyPI](https://pypi.org/project/scalex/) at the moment),
- [ ] add more tests for the model initialization and the model's performance,
- [ ] `src/` layout (see [cookiecutter-scverse-instance](https://github.com/scverse/cookiecutter-scverse-instance)) if you want to follow the template.

I don't think any of those are blocking for this PR so I'll merge this one.
